lib/caasp.pm

```diff
@@ -158,10 +158,27 @@ sub check_reboot_changes {
 sub trup_install {
     # rebootmgr has to be turned off as prerequisity for this to work
     script_run "rebootmgrctl set-strategy off";
-    my $package = shift;
-    trup_call("pkg install $package");
-    process_reboot 1;
-    assert_script_run("rpm -qi $package");
+    my ($input, $unnecessary, $necessary, $pkg_is_present) = shift;
+    my ($pkg_required, $not_needed) = 0;
+    my @pkg = split(' ', $input);
+    foreach (@pkg) {
```

Comment:
You can also encapsulate `split(' ', $input)` in foreach. Actually, as function gets list of packages, I would simply use array. In case you need more parameters, you can either use array reference, not to have issues with `@_`, or using hash.

Reply:
I originally started also with arrays because it *feels* the *right thing to do* for string manipulation. But something was not working as it should. I used `push` but the value in the end was `1` -- I don't know from where it came from. Anyway, I am not that experienced with Perl and this PR fixes a ticket with **high** priority. So, I would prefer to merge this code, open a ticket and assign to me to fix with with arrays next week, unless if you can *paste* me the code that works with arrays as an example I can happily give it a try. My problem is that time is pressing :/